infra/aws/terraform/artifacts.k8s.io/providers.tf

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+terraform {
+  backend "s3" {
+    bucket = "artifacts-k8s-io-tfstate"
+    key    = "terraform.tfstate"
+    region = "us-east-2"
+  }
+
+  required_version = "~> 1.2.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
```

Comment:
Do we want to lock the remote state? See https://developer.hashicorp.com/terraform/language/settings/backends/s3 for details.

Reply:
That could be a good idea, but we aren't doing it elsewhere, and it introduces a dependency on DynamoDB, so I suggest we add it separately.  TBH I'm not even sure how we apply this terraform for production, we might well have some protection against concurrent execution at a different layer.